astrosat_users/models/models_customers.py

```diff
@@ -149,10 +153,38 @@ def clean(self):
         if self.pk is None and self.user in self.customer.users.all():
             raise ValidationError("User is already a member of Customer.")
 
-    def invite(self):
-        # TODO: for now I'm just setting the date, need to actually do the invite
-        self.invitation_date = timezone.now()  # (passes a datetime object using the timezone from settings.py)
-        self.save()
+    def invite(self, **kwargs):
+        """
+        Makes the user aware that they are now a member of the customer
+        by sending an invitation email to their registered address.  If
+        they are a newly-created user, they will have to update their
+        password; the email will contain instructions on how to do this.
+        """
+
+        adapter = kwargs.get("adapter", get_adapter())
+        context = kwargs.get("context", {})
+        template_prefix = kwargs.get("template_prefix", "astrosat_users/email/invitation")
 
-# class CustomerInvitation(models.Model):
-#     pass
+        user = self.user
+        customer = self.customer
+        context.update({
+            "user": user,
+            "customer": customer,
+        })
+
+        if user.change_password:
+            token_generator = kwargs.get("token_generator", adapter.default_token_generator)
+            token_key = token_generator.make_token(user)
+            url = adapter.get_password_confirmation_url(adapter.request, user, token_key)
+            context["password_reset_url"] = url
+
+        if (
+            allauth_app_settings.AUTHENTICATION_METHOD
+            != allauth_app_settings.AuthenticationMethod.EMAIL
+        ):
+            context["username"] = user_username(user)
```

Comment:
Is there another authentication method? Or are you just future-proofing the code?

Reply:
**allauth** can have an authentication method of "username|email|username_email”; **orbis** happens to use "email"  but if an app ever required users to authenticate w/ their username I want to provide it in the invitation email.